### ISSUE TITLE
Understanding Non-text Contrast

### ISSUE THREAD

**Issue by michael-n-cooper** [w3c/wcag#346]
_From @awkawk on March 2, 2018 15:14_

<p>The   <a href="https://www.w3.org/TR/WCAG21/#non-text-contrast"   target="_blank">1.4.11 Non-text Contrast (AA) SC</a> has   draft Understanding content that can be viewed at <a   href="https://rawgit.com/w3c/wcag21/non-text-contrast/understanding/21/non-text-contrast.html"   target="_blank">Understanding Non-text   Contrast</a>.</p><p>Please review the Understanding   document and submit a comment on this issue for any changes that are needed   to the content.</p>

_Copied from original issue: w3c/wcag21#782_

**Comment by michael-n-cooper** [comment]
Understanding content for this issue should address the change agreed to by the WG in issue #805.

**Comment by michael-n-cooper** [comment]
_From @bocampbell on April 19, 2018 17:27_

Review Comments: 

Intent 
- P2 - "controls" is not defined. what is a control? 

Active User Interface Components 
- P2,3  what does it mean for a control to have or not have a visual indicator? 
- P4 this sentence isn't clear to me. visible indication of the hit area doesn't immediately mean anything to me. would this be the cursor turning into a hand? 

Active User Interface Component Examples
- P1 using the term "components" here when "controls" were used early. looking for a specific definition.
- P2 (after table) "html elements" is being used rather than control or component

Inactive User Interface Components
- I typically explain this to designers as a decision about whether the disabled component is meaningful in its disabledness so they can decide how visible to make them.

Graphical Objects
- These examples are very clear to me and very helpful.
- The graph could be simplified for explanation purposes and I'm not sure you need an example of lines on it that are NOT accessible to prove the points of what is and what is not a graphical object.
- The pie chart mentions a values being presented in a "conforming manner" and if it's not too long would be nice to understand one of those manners.
- The magnet explanation is good. It might be nice to have visual representations of other ways the magnet would be accessible, such as with just the tips outlined. This does make sense to me but I'm jaded.

Dynamic Examples
- This paragraph is not clear to me. I'm not sure what you're trying to say here. 

Essential [E]xception
- P1 The first part of this document explains that if a graphical object is important to the meaning of the icon, then it must be accessible. This paragraph starts out by saying that the exception is when the presentation of a graphic is essential to the information being conveyed. These things sound the same to me. I had to read this several times to understand why this new type of graphic was different from an icon or other "meaningful" images.
- P1 I guess I might have understood if you said something simpler like, "you are not required to alter a graphic if that alteration will change its meaning..." 
- Representing other things bullet: The word "essential" is emphasized here and I then had to search up in the first paragraph to see what that meant or why that was significant. I could see after doing that that the term, "essential," is the crux of this point. That wasn't clear at first. In other words, in this bullet, saying that the graphic is "essential" doesn't mean anything to me until I search for that meaning above and try to make that connection.

Examples
- I have had designers recently who know this constraint is coming ask me about using only an underline on a text box and not an entire box. I think that there will be a lot of questions like that coming up. How many sides constitutes a border?

**Comment by michael-n-cooper** [comment]
_From @lseeman on April 22, 2018 17:38_

Can we add that:
For people with cognitive disabilities, it is also recommended to use clear affordances, such that interactive controls are visually discriminable and have a border around the interactive area. Alternitvly  controls are available in a standard style that indicates how they can be used and what action it will trigger or have clearly labeled instructions that explains their use. 
@NeilMilliken any suggested changes?

**Comment by michael-n-cooper** [comment]
_From @MikeElledge on April 23, 2018 20:27_

P2 "Regardless of the whether or not there is a visible indication of the hit area for the button, the focus indicator for the component must have sufficient contrast when the component is focused." It might be helpful to mention in the first paragraph that this Success Criteria does not apply to the target (i.e., clickable) area of controls.

Also, like Bo, I don't understand the paragraph about Dynamic Controls.

**Comment by michael-n-cooper** [comment]
_From @mbgower on April 27, 2018 18:36_

@alastc , I'm incorporating folks' feedback, but I'm stumped in one place so far:
> like Bo, I don't understand the paragraph about Dynamic Controls.

could we link to that dynamic pie chart as an illustration of dynamic examples, or find a visual to incorporate?

This is really becoming a coherent piece of work overall, but I think the Understanding document could benefit from a whole lot more visual examples, even in the Intent section. It would really solidify understanding for many.

**Comment by michael-n-cooper** [comment]
_From @mbgower on April 27, 2018 18:57_

Submitted changes as https://github.com/w3c/wcag21/pull/885
@alastc there are some conflicts, it looks like with something you did with David's feedback. i'll see if I can figure out the issue.

**Comment by michael-n-cooper** [comment]
_From @alastc on April 28, 2018 0:17_

> I think the Understanding document could benefit from a whole lot more visual examples, even in the Intent section.

I'll take a look, but if you'd like to put in some suggestions I can try and get some design time. Our designers are **_much_** better & quicker than me if I can give a clear brief.

E.g. @@@ example of graphic showing X and Y.

**Comment by michael-n-cooper** [comment]
_From @mbgower on April 30, 2018 17:37_

Okay @alastc , here you go, https://github.com/w3c/wcag21/pull/888
I assigned you to review the PR

**Comment by michael-n-cooper** [comment]
_From @alastc on May 16, 2018 16:5_

Added pull request #924 which includes the graphics examples and does some re-structuring.

Mostly editorial and obvious graphics updates. Two key changes though:
- Removed the bit about relative luminance in the 1.4.1 section. I don't think it is an agreed interpretation that each state has to differ from the other states.
- Added a possibly controversial example (4) in the inputs table, similar to an example from the Material Design library.

Probably easiest to read in the [rendered version](https://rawgit.com/w3c/wcag21/non-text-contrast-ac/understanding/21/non-text-contrast.html).

I think I've included everyones changes, if any have been dropped they would be from David, as I had to merge that into Michael's version.

**Comment by alastc** [comment]
I think everything has either been incorporated or its changed a lot since then, closing.
